### PR TITLE
Add subscribe_invoice() example

### LINKED ISSUE #10: Add streaming example
Can I use the streaming gRPC methods? It would be nice to have an example, something like `SubscribeInvoices`.

### PR DESCRIPTION
**Pull request by grunch**
Fix #10

### CODE REVIEW

File: examples/subscribe_invoices.rs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+    let mut invoice_stream = client
+        .subscribe_invoices(tonic_lnd::rpc::InvoiceSubscription { add_index: 0, settle_index: 0 })
+        .await
+        .expect("Failed to call subscribe_invoices")
+        .into_inner();
+
+    while let Some(invoice) = invoice_stream.message().await.expect("Failed to receive invoices") {
+        // If this invoice was paid we can do something with it
+        if invoice.settle_date > 0 {
```
Comment (Kixunil) [inline examples/subscribe_invoices.rs:26]: Isn't checking `settled` field better? (I may be misremembereing something)

Comment (grunch) [reply examples/subscribe_invoices.rs:26]: Yeah I used that at first but turns out settled is deprecated

Comment (grunch) [reply examples/subscribe_invoices.rs:26]: If you don't like use `settle_date` we can use `state == "Open"`

Comment (grunch) [reply examples/subscribe_invoices.rs]: I think is a good idea, I fixed it to compare the state instead

File: examples/subscribe_invoices.rs [hunk 2]
```diff
@@ -0,0 +1,32 @@
+// This program accepts three arguments: address, cert file, macaroon file
+// The address must start with `https://`!
+
```
Comment (Kixunil) [inline examples/subscribe_invoices.rs:3]: It'd be nice to add some description for quick understanding. E.g.:

> This program connects to LND and prints out all incoming invoices as they settle.

File: examples/subscribe_invoices.rs [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        .subscribe_invoices(tonic_lnd::rpc::InvoiceSubscription { add_index: 0, settle_index: 0 })
+        .await
+        .expect("Failed to call subscribe_invoices")
+        .into_inner();
+
+    while let Some(invoice) = invoice_stream.message().await.expect("Failed to receive invoices") {
+        if let Some(state) = tonic_lnd::rpc::invoice::InvoiceState::from_i32(invoice.state) {
+            // If this invoice was Settled we can do something with it
+            if state == tonic_lnd::rpc::invoice::InvoiceState::Settled {
+                println!("{invoice:?}");
```
Comment (Kixunil) [inline examples/subscribe_invoices.rs:28]: Oh, just noticed this doesn't obey MSRV. I think it's fine for an example, just be mindful if you want to contribute into the library next time.

**Review by Kixunil — COMMENTED** [review]
The code looks good, just a small suggestion if you don't mind.

File: examples/subscribe_invoices.rs (hunk 3 above)
Comment (grunch) [reply examples/subscribe_invoices.rs:28]: what do you mean with `doesn't obey MSRV`? what part of the code? it works I tested and it works

File: examples/subscribe_invoices.rs (hunk 2 above)
Comment (grunch) [reply examples/subscribe_invoices.rs:3]: I followed you getinfo example which doesn't say  `This program connects to LND and prints out getinfo lnd node output`, you should update that one 😃

File: examples/subscribe_invoices.rs (hunk 3 above)
Comment (Kixunil) [reply examples/subscribe_invoices.rs:28]: Not writing the arguments in `println` was only stabilized recently but this crate supports Rust 1.48.0

File: examples/subscribe_invoices.rs (hunk 2 above)
Comment (Kixunil) [reply examples/subscribe_invoices.rs:3]: I know, I will. :) It just occurred to me now that the description is missing.

**Review by Kixunil — APPROVED** [review]
Thanks!

**Pull request merged**